src/baseband.c

```diff
@@ -116,7 +116,7 @@ float magnitude_true_cs16(int16_t const *iq_buf, uint16_t *y_buf, uint32_t len)
     for (i = 0; i < len; i++) {
         int32_t x = iq_buf[2 * i];
         int32_t y = iq_buf[2 * i + 1];
-        y_buf[i]  = (int)sqrt(x * x + y * y) >> 1; // max 46341, scaled 23170, fs 16384
+        y_buf[i]  = (int)sqrt((double)x * x + (double)y * y) >> 1; // max 46341, scaled 23170, fs 16384
```

Comment:
The code is fixed point by design.

Reply:
Yes, but `sqrt` takes a `double` which is why there is the warning.

Reply:
I think that sqrtf() should work and is slightly better type wise. If it is better accuracy wise I don't know.

Reply:
I guess the point is that x and y are limited to [-128,127] because of iq_buf being bytes, but the compiler is not figuring it out?   Is there in fact an overflow possibility, as determined by a human reviewing?  Or is it a false positive warning.   The position I have seen elsewhere, and tend to agree with, is that code should be fixed for valid warnings, and not modified much if any for invalid warnings, except that restructuring it slightly to be easier to reason about, withtout hurting the generated code is ok.